src/blueteam/Robot.java

```diff
@@ -131,4 +138,29 @@ boolean willCollideWithMe(BulletInfo bullet) {
 
 		return (perpendicularDist <= rc.getType().bodyRadius);
 	}
+
+	/**
+	 * Try to move in direction perpendicular to bullet trajectory
+	 *
+	 * @param bullet
+	 * @return true if move succeeded
+	 */
+	boolean trySidestep(BulletInfo bullet) {
+		Direction towards = bullet.getDir();
+		return (tryMove(towards.rotateRightDegrees(90)) || tryMove(towards.rotateLeftDegrees(90)));
+	}
+
+	/**
+	 * Try to dodge all bullets in range.
+	 */
+	void dodge() {
+		BulletInfo[] bullets = rc.senseNearbyBullets();
+		Arrays.stream(bullets).filter(x -> willCollideWithMe(x)).forEach(x -> trySidestep(x));
```

Comment:
The same algorithm twice.
But is it even correct? Can we move more than once in a single turn?
What if we move in range of another bullet?
I would write it such that we test e.g. 8 neighbouring location, count number of collisions using willCollideWithLocation() for each position and choose the position with least collisions... (including current location)